test-case/check-fw-echo-reference.sh

```diff
@@ -80,21 +71,24 @@ do
     printf "Testing: iteration %d of %d with %s format\n" "$i" "$loop_cnt" "$fmt"
         # S24_LE format is not supported
         if [ "$fmt" == "S24_LE" ]; then
+            dlogi "S24_LE is not supported, skip to test this format"
             continue
         fi
         # run echo reference test
         dlogc "alsabat -P $pb_dev --standalone -c $channel -f $fmt -r $rate -n $frames -F $frequency"
-        alsabat -P "$pb_dev" --standalone -c "$channel" -f "$fmt" -r "$rate" -n "$frames" -F "$frequency" &
-        # playback may have low latency, add 0.5 second delay to aviod recording zero at beginning.
+        alsabat -P "$pb_dev" --standalone -c "$channel" -f "$fmt" -r "$rate" -n "$frames" -F "$frequency" &\
+         alsabatPID=$!
+        # playback may have low latency, add 1 second delay to aviod recording zero at beginning.
         sleep 1
         dlogc "alsabat -C $cp_dev -c $channel -f $fmt -r $rate -F $frequency"
-        alsabat -C "$cp_dev" -c "$channel" -f "$fmt" -r "$rate" -F "$frequency"
+        alsabat -C "$cp_dev" -c "$channel" -f "$fmt" -r "$rate" -F "$frequency" || {
+            # upload failed wav files
+            find /tmp -maxdepth 1 -type f -name "bat.wav.*" -size +0 -exec cp {} "$LOG_ROOT/" \;
+            die "alsabat test failed on pcm: $cp_dev"
+        }
+
+        timeout 5 tail --pid=$alsabatPID -f /dev/null || die "Failed to stop alsabat playback"
```

Comment:
Sorry I gave you some wrong advice earlier: `timeout wait $pid` does not work because `wait` is a built-in that  works only for direct child processes. `tail --pid` is an interesting idea but it doesn't not collect the exit status/error code of the alsabat command which is the goal here so `tail --pid` doesn't not solve the problem. This works:

```shell
timeout -k 10 5 alsabat -P "$pb_dev" -... -F "$frequency" & timedAlsabatPID=$!
...
...
wait $timedAlsabatPID || die "alsabat failed..."
```
Unlike my previous advice I tested this logic and it does fail when alsabat fail.

Reply:
Thanks @marc-hb  for your suggestion, I have updated the code and did some fine-tuning of the time.